heron/tools/cli/src/python/restart.py

```diff
@@ -59,36 +59,36 @@ def run(command, parser, cl_args, unknown_args):
   :return:
   '''
   topology_name = cl_args['topology-name']
-  try:
-    container_id = cl_args['container-id']
+  container_id = cl_args['container-id']
 
-    new_args = [
-        "--cluster", cl_args['cluster'],
-        "--role", cl_args['role'],
-        "--environment", cl_args['environ'],
-        "--heron_home", config.get_heron_dir(),
-        "--config_path", cl_args['config_path'],
-        "--override_config_file", cl_args['override_config_file'],
-        "--release_file", config.get_heron_release_file(),
-        "--topology_name", topology_name,
-        "--command", command,
-        "--container_id", str(container_id)
-    ]
+  new_args = [
+      "--cluster", cl_args['cluster'],
+      "--role", cl_args['role'],
+      "--environment", cl_args['environ'],
+      "--heron_home", config.get_heron_dir(),
+      "--config_path", cl_args['config_path'],
+      "--override_config_file", cl_args['override_config_file'],
+      "--release_file", config.get_heron_release_file(),
+      "--topology_name", topology_name,
+      "--command", command,
+      "--container_id", str(container_id)
+  ]
 
-    lib_jars = config.get_heron_libs(jars.scheduler_jars() + jars.statemgr_jars())
+  if Log.getEffectiveLevel() == logging.DEBUG:
+    new_args.append("--verbose")
 
-    # invoke the runtime manager to kill the topology
-    execute.heron_class(
-        'com.twitter.heron.scheduler.RuntimeManagerMain',
-        lib_jars,
-        extra_jars=[],
-        args=new_args
-    )
+  lib_jars = config.get_heron_libs(jars.scheduler_jars() + jars.statemgr_jars())
 
-  except Exception as ex:
-    Log.debug(traceback.format_exc(ex))
+  # invoke the runtime manager to kill the topology
+  retcode = execute.heron_class(
+      'com.twitter.heron.scheduler.RuntimeManagerMain',
+      lib_jars,
+      extra_jars=[],
+      args=new_args
+  )
+  if retcode != 0:
     Log.error('Failed to restart topology \'%s\'' % topology_name)
     return False
-
-  Log.info('Successfully restarted topology \'%s\'' % topology_name)
-  return True
+  else:
+    Log.info('Successfully restarted topology \'%s\'' % topology_name)
```

Comment:
2. Current design requires each command do this at the end:
```
if retcode != 0:
   Log.error('Failed to update topology \'%s\'', topology_name)
   return False
```

So we only have a consistent solution if every command implement cargo-cults just right. Better to codify this into the response time that is required to be returned and let a single agent implement the logging logic. That way it will happen consistently, even if we decide to change some output to go to `print` instead of `Log` for instance, or if we decide to render exception or verbose responses differently at some point.

It's kind of an extension of the log-and-return-null|false pattern that we're trying to remove in the java code.